File: src/main/java/io/github/aphelium/modules/brokers/entities/BrokerCoins.java
package io.github.aphelium.modules.brokers.entities;

import io.github.aphelium.modules.coin.entities.Coin;
import lombok.Data;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;
import java.util.UUID;

@Data
@Entity
@Table(name = "TB_BROKER_COINS")
public class BrokerCoins implements Serializable {

    @Serial
    private static final long serialVersionUID = -8080540494839892473L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false)
    private UUID broker;

    @Column(nullable = false)
    private UUID coin;

}

File: src/main/kotlin/io/github/aphelium/modules/brokers/entities/BrokerCoins.kt
package io.github.aphelium.modules.brokers.entities

import java.io.Serial
import java.io.Serializable
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "TB_BROKER_COINS")
class BrokerCoins : Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: UUID? = null

    @Column(nullable = false)
    var broker: UUID? = null

    @Column(nullable = false)
    var coin: UUID? = null

    companion object {
        @Serial
        private val serialVersionUID = -8080540494839892473L
    }
}